assets/app.js

```diff
@@ -338,6 +338,39 @@ new Vue({
         }
     },
     computed: {
+        validateCacheMaterial: function () {
+            active_languages = this.activeLanguagesId
+            this.products.forEach(function (product) {
+                active_languages.forEach(function (language) {
+                    materials = product.materials.map(function (material) {
+                        return material[language].trim()
+                    })
+
+                    cached_materials = product.cached_materials[language].value.split('/').map(function(resource){
+                        return resource.trim();
+                    });
+
+                    is_overridden = true
+                    if (materials.length === 0) {
+                        is_overridden = false
+                    }
+                    //compare cached_materials and materials
+                    if (cached_materials.length === materials.length) {
+                        is_overridden = false
+                        for (i = 0; i < materials.length; i++) {
```

Comment:
is there a reason for `for (i = 0; i < materials.length; i++)` instead `materials.foreach(...)`?

Reply:
`foreach` cannot use `break`

Reply:
maybe you'd like `materials.some` not sure what's it doing, so.

Reply:
this is a very complicated way to figure out if something changed. I don't think this works properly.
What about
```
// filter active materials
current_materials_list = product.materials.map(function (material) {
   if(material.is_active) return material[language]
}
// join materials to preview what we'll write back to DB
current_materials = current_materials_list.join("/") 
product.cached_materials[language]['is_overridden'] = (product.cached_materials[language]=== current_materials)